src/latbin/lattice-to-fst.cc

```diff
@@ -60,31 +124,65 @@ int main(int argc, char *argv[]) {
     std::string lats_rspecifier = po.GetArg(1),
         fsts_wspecifier = po.GetArg(2);
     
-    SequentialCompactLatticeReader lattice_reader(lats_rspecifier);
+    TransitionModel tmodel;
+    if (!trans_model.empty()) {
+      ReadKaldiObject(trans_model, &tmodel);
+    }
+
+    SequentialCompactLatticeReader compact_lattice_reader;
+    SequentialLatticeReader lattice_reader;
+
     TableWriter<fst::VectorFstHolder> fst_writer(fsts_wspecifier);
     
     int32 n_done = 0; // there is no failure mode, barring a crash.
     
-    for (; !lattice_reader.Done(); lattice_reader.Next()) {
-      std::string key = lattice_reader.Key();
-      CompactLattice clat = lattice_reader.Value();
-      lattice_reader.FreeCurrent();
-      ScaleLattice(scale, &clat); // typically scales to zero.
-      RemoveAlignmentsFromCompactLattice(&clat); // remove the alignments...
-      fst::VectorFst<StdArc> fst;
-      {
-        Lattice lat;
-        ConvertLattice(clat, &lat); // convert to non-compact form.. won't introduce
-        // extra states because already removed alignments.
-        ConvertLattice(lat, &fst); // this adds up the (lm,acoustic) costs to get
-        // the normal (tropical) costs.
-        Project(&fst, fst::PROJECT_OUTPUT); // Because in the standard Lattice format,
-        // the words are on the output, and we want the word labels.
+    if (read_compact) {
+      SequentialCompactLatticeReader compact_lattice_reader(lats_rspecifier);
+      for (; !compact_lattice_reader.Done(); compact_lattice_reader.Next()) {
+        std::string key = compact_lattice_reader.Key();
+        CompactLattice clat = compact_lattice_reader.Value();
+        compact_lattice_reader.FreeCurrent();
+        ScaleLattice(scale, &clat); // typically scales to zero.
+        RemoveAlignmentsFromCompactLattice(&clat); // remove the alignments...
+        fst::VectorFst<StdArc> fst;
+        {
+          Lattice lat;
+          ConvertLattice(clat, &lat); // convert to non-compact form.. won't introduce
```

Comment:
I don't like that these multi-line comments have different indentation on different lines.